src/simplify_qdq.cpp

```diff
@@ -0,0 +1,205 @@
+    std::vector<float> scales;
+    scale_lit.visit([&](auto sl) {
+        std::transform(
+            sl.begin(), sl.end(), std::back_inserter(scales), [](auto&& s) { return s; });
+    });
+    double epsilon = 1e-6;
+    if(not std::all_of(scales.begin(), scales.end(), [&](const auto& s) {
+           return std::abs(s - scales.front()) < epsilon;
+       }))
+        MIGRAPHX_THROW("Multiple scales not currently supported");
+    return scales.front();
+}
+
+instruction_ref insert_quantize_op(module& m,
+                                   instruction_ref ins,
+                                   const std::string& name,
+                                   instruction_ref x,
+                                   instruction_ref scale,
+                                   instruction_ref shift)
+{
+    auto lens = x->get_shape().lens();
+    auto scale_mb =
+        m.insert_instruction(ins, make_op("multibroadcast", {{"output_lens", lens}}), scale);
+    auto shift_mb =
+        m.insert_instruction(ins, make_op("multibroadcast", {{"output_lens", lens}}), shift);
+    return m.insert_instruction(ins, make_op(name), x, scale_mb, shift_mb);
+}
+
+struct match_find_quantizable_ops
+{
+    auto matcher() const
+    {
+        return match::name(get_all_op_names())(match::arg(0)(
+            match::name(get_quantizable_op_names())(
+                match::arg(0)(match::name("dequantizelinear")(
+                    match::arg(0)(match::name(get_all_op_names()).bind("q0")).bind("dq0"))),
+                match::arg(1)(match::name("dequantizelinear")(
+                    match::arg(0)(match::name(get_all_op_names()).bind("q1")).bind("dq1"))))
+                .bind("qop")));
+    }
+
+    void apply(module& m, match::matcher_result r) const
+    {
+        auto op  = r.result;
+        auto qop = r.instructions["qop"];
+        auto dq0 = r.instructions["dq0"];
+        auto q0  = r.instructions["q0"];
+        auto dq1 = r.instructions["dq1"];
+        auto q1  = r.instructions["q1"];
+
+        // Only INT8 type currently supported
+        if(q0->get_shape().type() != migraphx::shape::int8_type or
+           q1->get_shape().type() != migraphx::shape::int8_type)
+            return;
+
+        // Only zero_point==0 currently supported
+        auto dq0_args  = dq0->inputs();
+        auto dq1_args  = dq1->inputs();
+        bool all_zeros = true;
+        all_zeros &= inputs_are_zeros(dq0_args.at(2));
+        all_zeros &= inputs_are_zeros(dq1_args.at(2));
+        if(not all_zeros)
+            return;
+
+        auto scale     = get_scale(dq0_args.at(1)) * get_scale(dq1_args.at(1));
+        auto qop_args  = qop->inputs();
+        qop_args.at(0) = q0;
+        qop_args.at(1) = q1;
+        instruction_ref dq;
+        instruction_ref dq_scale;
+        instruction_ref zero_point;
+        if(qop->name() == "convolution")
+        {
+            auto conv_op = any_cast<op::convolution>(qop->get_operator());
```

Comment:
This can be done in a simpler way as:

```
auto conv_val = qop->get_operator().to_value();
qop          = m.insert_instruction(qop, migraphx::make_op("quant_convolution", conv_val), qop_args);
```